processors/base/writers/location.py

```diff
@@ -70,3 +76,27 @@ def write_location(conn, location, source_id, trial_id=None):
             'created' if create else 'updated', name)
 
     return object['id']
+
+#----helper function to get the canonical name of the locations
+def extract_CanonicalName(location):
+    location_name = location.lower().strip().replace(".", "")
+    try:
+        canonical_name = countries.get(location_name).name
+        logger.debug('Location - %s: %s','normalized', canonical_name)
+    except KeyError:
+        canonical_name = location
+        # This part is for looking for matching cases (suggestion), uses the 
+        # fuzzy logic to get the country name who get 80% of match rate.
+        with open(os.path.join(os.path.dirname(__file__),'countries.csv'), 'r') as f:
+            countries_data = csv.reader(f, delimiter= str(u','))
+            logger.debug('Location - %s: %s','not normalized', canonical_name)
+            best_similarity = 0
+            for country in countries_data:
+                unicode_row = [x.decode('utf8') for x in country]
+                country_name = unicode_row[0].lower().strip().replace(".", "")
+                if ((fuzz.ratio(location_name, country_name) >= 80) and 
+                    (fuzz.ratio(location_name, country_name) > best_similarity)):
+                    canonical_name = countries.get(unicode_row[3]).name
+                    logger.debug('Location - %s: %s','not normalized with the most close result', canonical_name)
```

Comment:
The `best_similarity` is never updated. To avoid doing these two `fuzz.ratio` calls, you could simply set `best_similarity = 80` and then just test if `fuzz.ratio(...) >= best_similarity`, updating the `best_similarity` value.